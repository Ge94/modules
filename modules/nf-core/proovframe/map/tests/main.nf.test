nextflow_process {

    name "Test Process PROOVFRAME_MAP"
    script "../main.nf"
    process "PROOVFRAME_MAP"

    tag "modules"
    tag "modules_nfcore"
    tag "proovframe"
    tag "proovframe/map"

    test("sarscov2 - provided db") {

        when {
            process {
                """
                input[0] = [
                    [ id:'fa', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/proteome.fasta', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + 'proteomics/database/UP000005640_9606.dmnd', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.tsv[0][1]).text.contains('MT192765.1	sp|Q460N5|PAR14_HUMAN	28.4	176	105	5	3424	3918	818	983') },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

    test("sarscov2 - provided faa") {

        when {
            process {
                """
                input[0] = [
                    [ id:'fa', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/proteome.fasta', checkIfExists: true)
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.tsv[0][1]).text.contains('MT192765.1	ENSSASP00005000002.1	100.0	7096	1	0	259	21545	1') },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

    test("sarscov2 - tsv - stub") {
        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test2', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/proteome.fasta', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + 'proteomics/database/UP000005640_9606.dmnd', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
