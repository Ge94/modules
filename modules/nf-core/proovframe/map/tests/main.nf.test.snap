{
    "sarscov2 - provided faa": {
        "content": [
            [
                "versions.yml:md5,f49f29b8134fdb539364ee46855eb313"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-10-27T13:13:01.777154464"
    },
    "sarscov2 - provided db": {
        "content": [
            [
                "versions.yml:md5,f49f29b8134fdb539364ee46855eb313"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-10-27T13:10:35.425578608"
    },
    "sarscov2 - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        "test2.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f49f29b8134fdb539364ee46855eb313"
                ],
                "tsv": [
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        "test2.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f49f29b8134fdb539364ee46855eb313"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-10-27T13:15:18.917890463"
    }
}